strawberry/persistance/http.py

```diff
@@ -0,0 +1,32 @@
+from abc import ABC
+from typing import Dict
+
+from typing_extensions import TypedDict
+from werkzeug.datastructures import ImmutableMultiDict
+
+from flask import request
+
+
+class ExtensionData(TypedDict):
+    """
+    The data from the incoming extensions
+    """
+
+    version: int
+    sha256Hash: str
+
+
+class RequestPopulator(ABC):
+    def populate(self, data: Dict):
+        """Populates the Flask request with dictionary-data"""
+        raise NotImplementedError("Method should be implemented in subtypes")
+
+
+class RequestJsonPopulator(RequestPopulator):
+    def populate(self, data: Dict):
+        request._cached_json = (data, data)
```

Comment:
I'm not sure this is somethign we should do, using a private attribute to return JSON sounds wrong, is there a particular reason why we don't return JSON directly?

Reply:
Regarding my previous comment, the default Flask view initially checks for data in the `request.json` dict. In this case, this always gets called, unless when a custom view is defined. That's why re-populating the request json is nessacery in this case.